src/pages/Flavors.test.js

```diff
@@ -139,126 +152,92 @@ describe('Page <Flavors />', () => {
       }
     }
   ];
-  const pager = { count: 5, limit: 20, page: 1, pages: 1 };
-  const store = mockStore({
-    actions,
-    application: appInitialState,
-    flavor: flavorInitialState,
-    flavors: flavorsInitialState
-  });
 
+  const pager = { count: 5, limit: 20, page: 1, pages: 1 };
   const pagerNavigation = <p></p>;
 
-  const notLoggedInProps = {
-    actions,
+  const props = {
     collection,
-    loggedIn: false,
     pager,
-    pagerNavigation,
-    stash,
-    stashLoaded: false
+    pagerNavigation
   };
 
-  const noStashProps = {
-    actions,
-    collection,
-    loggedIn: true,
-    pager,
-    pagerNavigation,
-    stash,
-    stashLoaded: false
-  };
+  const ReduxConnectedFlavors = (flavorsProps) => (
+    <Provider store={store}>
+      <Flavors {...flavorsProps} />
+    </Provider>
+  );
+
+  beforeEach(() => {
+    useDispatch().mockClear();
+    isLoggedIn.mockClear();
+    getStash.mockClear();
+    isLoaded.mockClear();
+  });
 
-  const props = {
-    actions,
-    collection,
-    loggedIn: true,
-    pager,
-    pagerNavigation,
-    stash,
-    stashLoaded: true
-  };
+  it('has "log in" call to action when user is not logged in', () => {
+    isLoggedIn.mockReturnValue(false);
+    getStash.mockReturnValue([]);
+    isLoaded.mockReturnValue(false);
 
-  const RoutedConnectedFlavors = withMemoryRouter(ConnectedFlavors);
-  const RoutedFlavors = withMemoryRouter(Flavors);
+    const { getByText } = render(<ReduxConnectedFlavors {...props} />);
 
-  it('renders correctly', () => {
-    expect(
-      renderer
-        .create(
-          <Provider store={store}>
-            <RoutedConnectedFlavors />
-          </Provider>
-        )
-        .toJSON()
-    ).toMatchSnapshot();
+    expect(getByText(/log in/i)).toBeDefined();
   });
 
-  it('renders correctly when stash is not loaded', () => {
-    expect(
-      renderer.create(<RoutedFlavors {...noStashProps} />).toJSON()
-    ).toMatchSnapshot();
-  });
+  it('matches snapshot when stash is not loaded', () => {
+    isLoggedIn.mockReturnValue(true);
+    getStash.mockReturnValue([]);
+    isLoaded.mockReturnValue(false);
 
-  it('renders correctly when user is not logged in', () => {
-    expect(
-      renderer.create(<RoutedFlavors {...notLoggedInProps} />).toJSON()
-    ).toMatchSnapshot();
+    const { asFragment } = render(<ReduxConnectedFlavors {...props} />);
+
+    expect(asFragment()).toMatchSnapshot();
   });
 
-  const component = renderer.create(<RoutedFlavors {...props} />);
+  it("matches snapshot diff of user's stash (after toggle)", () => {
```

Comment:
I don't understand what this test does based on this name. I think it "ensures toggle button click updates state"?

Reply:
It toggles the button, and takes a snapshot of the difference between the html before and after the toggle